lua/apisix/plugins/prometheus/exporter.lua

```diff
@@ -44,26 +48,35 @@ function _M.init()
     -- per service
     metrics.status = prometheus:counter("http_status",
             "HTTP status codes per service in Apisix",
-            {"code", "service"})
+            {"code", "service", "node"})
+
+    metrics.latency = prometheus:histogram("http_latency",
+        "HTTP request latency per service in Apisix",
+        {"type", "service", "node"}, DEFAULT_BUCKETS)
 
     metrics.bandwidth = prometheus:counter("bandwidth",
             "Total bandwidth in bytes consumed per service in Apisix",
-            {"type", "service"})
+            {"type", "service", "node"})
 end
 
 
 function _M.log(conf, ctx)
     core.table.clear(tmp_tab)
 
-    local host = ctx.var.host
-    core.table.set(tmp_tab, ctx.var.status, host)
-    metrics.status:inc(1, tmp_tab)
+    local service_name
+    if ctx.matched_route and ctx.matched_route.value then
+        service_name = ctx.matched_route.value.desc or ctx.matched_route.value.id
+    end
+
+    local balancer_ip = ctx.balancer_ip
+    metrics.status:inc(1, {ctx.var.status, service_name, balancer_ip})
```

Comment:
Sorry. I'm a little confused by these two comments, so we should reuse table to avoid table temporary variables?

Reply:
This way you can reuse table to avoid table temporary variables.

```lua
core.table.set(tmp_tab, ctx.var.status, service_name, balancer_ip)
metrics.status:inc(1, tmp_tab)
```

Reply:
```lua
metrics.status:inc(1, {ctx.var.status, service_name, balancer_ip})
change to->
metrics.status:inc(1, ctx.var.status, service_name, balancer_ip)
```

The new style is better, we can change it later.